visual_behavior/data_access/loading.py

```diff
@@ -436,28 +463,40 @@ def get_extened_stimulus_presentations(stimulus_presentations, licks, rewards, r
     stimulus_presentations['pre_omitted'] = stimulus_presentations['omitted'].shift(-1)
     stimulus_presentations = reformat.add_epoch_times(stimulus_presentations)
     stimulus_presentations = reformat.add_mean_running_speed(stimulus_presentations, running_speed)
-    try:  # if eye tracking data is not present or cant be loaded
-        stimulus_presentations = reformat.add_mean_pupil_area(stimulus_presentations, eye_tracking)
-    except BaseException:  # set to NaN
-        stimulus_presentations['mean_pupil_area'] = np.nan
+    if eye_tracking:
+        try:  # if eye tracking data is not present or cant be loaded
+            stimulus_presentations = reformat.add_mean_pupil_area(stimulus_presentations, eye_tracking)
+        except BaseException:  # set to NaN
+            stimulus_presentations['mean_pupil_area'] = np.nan
     stimulus_presentations = reformat.add_licks_each_flash(stimulus_presentations, licks)
     stimulus_presentations = reformat.add_response_latency(stimulus_presentations)
     stimulus_presentations = reformat.add_rewards_each_flash(stimulus_presentations, rewards)
     stimulus_presentations['licked'] = [True if len(licks) > 0 else False for licks in
                                         stimulus_presentations.licks.values]
+    # lick rate per second
     stimulus_presentations['lick_rate'] = stimulus_presentations['licked'].rolling(window=320, min_periods=1,
                                                                                    win_type='triang').mean() / .75
-    stimulus_presentations['rewarded'] = [True if len(rewards) > 0 else False for rewards in
-                                          stimulus_presentations.rewards.values]
+    stimulus_presentations['rewarded'] = [True if len(rewards) > 0 else False for rewards in stimulus_presentations.rewards.values]
+    # (rewards/stimulus)*(1 stimulus/.750s) = rewards/second
+    stimulus_presentations['reward_rate_per_second'] = stimulus_presentations['rewarded'].rolling(window=320, min_periods=1,
+                                                                                                  win_type='triang').mean() / .75  # units of rewards per second
+    # (rewards/stimulus)*(1 stimulus/.750s)*(60s/min) = rewards/min
     stimulus_presentations['reward_rate'] = stimulus_presentations['rewarded'].rolling(window=320, min_periods=1,
-                                                                                       win_type='triang').mean()
+                                                                                       win_type='triang').mean() * (60 / .75)  # units of rewards/min
+
+    reward_threshold = 2  # threshold of 2 rewards per minute; if using rewards per second, use 1/90
```

Comment:
1/90 (rewards/s) = 2/3 (rewards/min), not 2 rewards/min.

If you want to use 2 rewards/min we can discuss the threshold, but Doug and I settled on 1/90 rewards/sec (2/3 rewards/min)

Reply:
changed to 2/3 rewards/min